src/sentry/utils/snuba.py

```diff
@@ -701,25 +701,34 @@ def nest_groups(data, groups, aggregate_cols):
         )
 
 
-def resolve_column(col, dataset):
-    if col.startswith("tags["):
-        return col
+def resolve_column(dataset):
+    def _resolve_column(col):
+        if col is None or col.startswith("tags[") or QUOTED_LITERAL_RE.match(col):
+            return col
+
+        # Some dataset specific logic:
+        if dataset == Dataset.Discover:
+            if isinstance(col, (list, tuple)) or col == "project_id":
+                return col
+        else:
+            if (
+                col in DATASET_FIELDS[dataset]
+            ):  # Discover does not allow internal aliases to be used by customers, so doesn't get this logic.
```

Comment:
Do you need access to internal schema names in alerts? Issues unfortunately uses this behavior already.

Reply:
We mostly just need the same set of column names that Discover uses, we're just unable to use the Discover dataset.

Reply:
At this point there are only two datasets so I'm not sure if I could easily distinguish between "alerts" and "issues" without introducing some tertiary option. 

Though, since we are querying the events table and users would likely expect the same query bar searches to work, shouldn't we support the same set of column names as issues?

Reply:
We don't need to worry about issues or their fields in metric alerts, just things we have access to in Snuba.

Reply:
Will you folks be re-adding the transactions dataset aliases `Columns`? Right now that translation layer is missing.

Reply:
I think we might need to, but we can address that in the follow up pr that we use this in.